types/gh-pages/index.d.ts

```diff
@@ -3,8 +3,27 @@
 // Definitions by: Daniel Rosenwasser <https://github.com/DanielRosenwasser>
 //                 Piotr Błażejewicz <https://github.com/peterblazejewicz>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-
+export interface Git {
+    cwd: string;
+    cmd: string;
+    output: string;
+    exec: (command: string) => Promise<this | Error>;
```

Comment:
```suggestion
    exec: (command: string) => Promise<this>;
```

Reply:
those methods resolve with results or reject with errors, e.g.:
https://github.com/tschaub/gh-pages/blob/6c790a004c9a1c4216d088c9736c7c8e6cb48e26/lib/git.js#L42
so I think those shoudl be only `Promise<this>`